### PR TITLE
Added HTTP2 headers support

### PR DESCRIPTION
**Pull request by webdevium**
#### Checklist

- [x] run `yarn test`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by webdevium** [comment]
> Could you add a unit test?

Yeap. One question: what I must test in unit?

**Comment by Eomm** [comment]
We need to do an HTTP2 call to the server and, in the handler of the route, we must verify that the parsed data is correct.
This will ensure that we are reading the header in the right way also for major releases of fastify

This is test need https, and here you can find an example:
https://github.com/fastify/fastify/blob/master/test/http2/secure.js

**Comment by webdevium** [comment]
Test added. But **h2url** not supports _path_ and _query_ in requests :(

And I need some help with CI @Eomm. Help me with Node v6

**Comment by webdevium** [comment]
I changed http2 client and added skip for test in v6 environment

File: test/tests.test.js [hunk 1]
```diff
@@ -41,3 +45,50 @@ test('parses a full URI', (t) => {
...
+    t.is(req.urlData('query'), 'a=b&c=d')
+    reply.send()
+  })
+
+  fastify.listen(0, (err) => {
+    fastify.server.unref()
+    if (err) t.threw(err)
+
+    port = fastify.server.address().port
+    h2client.request(`https://127.0.0.1:${port}/one?a=b&c=d#foo`, () => process.exit())
```
Comment (mcollina) [inline test/tests.test.js]: Why the `process.exit`?

**Comment by webdevium** [comment]
Without this string test stuck and stop only by tap’s timeout
t.end() not help too

**Comment by mcollina** [comment]
Can you try switching to h2url or node core http2? I fear that http2-client is leaving some state around (the http2 session).

**Comment by webdevium** [comment]
We must test basic H2 headers (node core functionality), not client :)

**Comment by webdevium** [comment]
I really don’t know how close h2 connection in any h2 clients. So, t.end() not work while connection is opened. Any ideas?

**Comment by mcollina** [comment]
Take a look at https://github.com/fastify/fastify/blob/master/test/http2/secure.js.

**Comment by webdevium** [comment]
I see this example. But h2url don’t support path and query in urls without pull request from @eomm

**Comment by webdevium** [comment]
I’ll try test with h2url tomorrow. Thanks!

**Comment by mcollina** [comment]
That was released yesterday.

Il giorno dom 20 ott 2019 alle 12:50 Vano Devium <notifications@github.com>
ha scritto:

**Comment by webdevium** [comment]
@mcollina Review, please.

**Comment by Eomm** [comment]
Implement #7

**Pull request merged**